src/crimson/os/seastore/transaction_manager.h

```diff
@@ -385,17 +385,17 @@ class TransactionManager : public ExtentCallbackInterface {
         std::vector<remap_entry>(remaps.begin(), remaps.end()),
         [this, &t, original_laddr, original_paddr,
 	original_len, intermediate_base, intermediate_key]
-        (auto &ret, auto &count, auto &original_bptr, auto &remaps) {
+        (auto &ret, auto &count, auto &original_bptr, auto &remaps) mutable {
         return _dec_ref(t, original_laddr, false
         ).si_then([this, &t, &original_bptr, &ret, &count,
 		   &remaps, intermediate_base, intermediate_key,
-                   original_laddr, original_paddr, original_len](auto) {
+                   original_laddr, original_paddr, original_len](auto) mutable {
           return trans_intr::do_for_each(
             remaps.begin(),
             remaps.end(),
             [this, &t, &original_bptr, &ret,
 	    &count, intermediate_base, intermediate_key,
-	    original_laddr, original_paddr, original_len](auto &remap) {
+	    original_laddr, original_paddr, original_len](auto &remap) mutable {
```

Comment:
How about reverting the above `mutable` keywords and introduce a `remapped_intermediate_key` below ?

Reply:
I didn't see the captures difference, so which variable is not read only and cause mutable needed?

Reply:
Done.